moveit_core/constraint_samplers/src/default_constraint_samplers.cpp

```diff
@@ -507,6 +509,15 @@ bool constraint_samplers::IKConstraintSampler::samplePose(Eigen::Vector3d& pos,
     quat = Eigen::Quaterniond(ikq.rotation());
   }
 
+  if (need_eef_to_ik_tip_transform_)
+  {
+    // After sampling the pose needs to be transformed to the ik chain tip
+    Eigen::Affine3d ikq(Eigen::Translation3d(pos) * quat.toRotationMatrix());
+    ikq = ikq * eef_to_ik_tip_transform_;
+    pos = ikq.translation();
+    quat = Eigen::Quaterniond(ikq.rotation());
+  }
```

Comment:
I think the [sampleHelper](http://docs.ros.org/hydro/api/moveit_core/html/default__constraint__samplers_8cpp_source.html#l00492) function is a more appropriate place for applying the transform.